### ISSUE TITLE
Unable to reproduce results on Cityscapes

### ISSUE THREAD

**Issue by benearnthof** [MICV-yonsei/EAGLE#7]
After running the provided preprocessing and training scripts as they are provided in this repository I was unable to replicate the results of EAGLE on the cityscapes dataset. I trained with the configs provided here and adjusted the training hyperparameters to those available through downloading the cityscapes checkpoint the authors provided through google drive. Even after 25000 training steps on cityscapes cluster test accuracy only reaches 67%. I have attached an output plot to this issue. Could you provide insight into how to replicate the results? I've noticed that there are additional clustering parameters present in the provided checkpoint that are not used in the train config files here. Do you perform an addtional post processing step?   

This is what my results look like after 25k training steps on cityscapes.  

<img src="https://github.com/user-attachments/assets/92235665-35ba-4890-b9ea-0feb7722a6c9" width=50% height=50%>

**Comment by benearnthof** [comment]
That must be what I have been missing, but I had assumed CRF would not make that much of a difference. I'll report back after I run the eval script. Thank you very much for the swift reply!

**Comment by benearnthof** [comment]
After running the evaluation on my own cityscapes checkpoint I obtain the following metrics:  

```python
{
'final/linear/mIoU': 32.06715285778046, 
'final/linear/Accuracy': 90.96953272819519, 
'assignments': [9, 8, 4, 6, 14, 5, 7, 11, 3, 18, 16, 26, 20, 12, 22, 23, 0, 1, 24, 10, 19, 25, 13, 15, 21, 2, 17], 
'final/cluster/mIoU': 15.385963022708893,
'final/cluster/Accuracy': 73.27690720558167
}
```  

Seems like the CRF postprocessing does a lot of heavy lifting I'll rerun training with 3000 steps like you recommended and report back. Thanks a lot for the help!

**Comment by benearnthof** [comment]
I've done evaluation on 5 other checkpoints that each were trained for 5000 steps where I picked the highest performing checkpoint for each run and ran them through the evaluation script. (Each training run saved checkpoints every 10 steps as was suggested to me in another issue.)  The mean cluster Accuracy for Cityscapes is 70.4, the maximum Accuracy I obtained after CRF evaluation was 74.1. A lot better than my previous results but still quite far from the performance reported in the paper. Did you do any additional hyperparameter tuning?

**Comment by evanchar** [comment]
> I've done evaluation on 5 other checkpoints that each were trained for 5000 steps where I picked the highest performing checkpoint for each run and ran them through the evaluation script. (Each training run saved checkpoints every 10 steps as was suggested to me in another issue.) The mean cluster Accuracy for Cityscapes is 70.4, the maximum Accuracy I obtained after CRF evaluation was 74.1. A lot better than my previous results but still quite far from the performance reported in the paper. Did you do any additional hyperparameter tuning?

Hello! I am trying to train a model but I am too unable to reproduce the results. Did you find any solutions and / or hyperparameters to get closer to the performance reported in the paper?

**Comment by kochanha** [comment]
Hi.
Try using W_comb =  W_color in eigen_modules.py #L257.

**Comment by evanchar** [comment]
> Hi. Try using W_comb = W_color in eigen_modules.py #L257.

I will try and report back, thanks!

**Comment by evanchar** [comment]
> Hi. Try using W_comb = W_color in eigen_modules.py #L257.

Hello again!
I tried the to change the code as you instructed but unfortunately didn't see any improvement. 

Also, I checked the configuration with which the checkpoint you provide for evaluation was trained, and I observed some differences with the config.yaml for cityscapes on github. Tried to also train a new model with that config, but still couldn't reproduce the checkpoint that you provided.

Could it be possible that the source training code got accidentally altered in some way, after the training that yielded your best checkpoint? 

Thanks!

**Comment by kochanha** [comment]
While cleaning up unnecessary code, there may have been an influence from the random seed. In unsupervised methods, the random seed plays a critical role in determining the initial centroids during the clustering process, making it a significant factor. I will check on this and get back to you.
